iroha_actor/src/envelope.rs

```diff
@@ -0,0 +1,57 @@
+#![allow(clippy::module_name_repetitions)]
+
+use async_std::sync as oneshot;
+
+use super::*;
+
+pub trait ToEnvelope<A: Actor + Handler<M>, M: Message> {
+    fn pack(msg: M, tx: Option<oneshot::Sender<M::Result>>) -> Envelope<A>;
+}
+
+pub struct Envelope<A: Actor>(pub Box<dyn EnvelopeProxy<A> + Send>);
+
+impl<A, M> ToEnvelope<A, M> for SyncEnvelopeProxy<M>
+where
+    A: Actor + Handler<M>,
+    M: Message + Send + 'static,
+    M::Result: Send,
+{
+    fn pack(msg: M, tx: Option<oneshot::Sender<M::Result>>) -> Envelope<A> {
+        Envelope(Box::new(Self { msg: Some(msg), tx }))
+    }
+}
+
+#[async_trait::async_trait]
+pub trait EnvelopeProxy<A: Actor> {
+    async fn handle(&mut self, act: &mut A, ctx: &mut Context<A>);
+}
+
+pub struct SyncEnvelopeProxy<M>
+where
+    M: Message + Send,
+    M::Result: Send,
+{
+    msg: Option<M>,
+    tx: Option<oneshot::Sender<M::Result>>,
```

Comment:
Please use full names. Also `tx` in our context is kind of reserved for transaction.